### ISSUE TITLE
Org Group tags are not recognized

### ISSUE THREAD

**Issue by k-jell** [alphapapa/org-super-agenda#136]
I'm using Org Group-Tags as described here: [Org Tag hierarchy.](https://orgmode.org/manual/Tag-hierarchy.html)

This lets me search for for a main Group-Tag via `org-tags-view` and it then displays all members of the group and any subgroups.

However `super-agenda` doesn't match the tags correctly. I'm not quite sure how that could be achieved as I didn't find a way to actually get the group tags. When I had a quick look at the `org-tags-view defun` it seems like it uses `org-make-tags-matcher` to match the group tags: [See source code here](https://github.com/tkf/org-mode/blob/c2ebeea6f68f2ef804d387c238e4acccf655dc64/lisp/org-agenda.el#L4626)

Anyway my Lisp is not very good so I'm not sure if that's even correct or if this can be used to make `super-agenda` match Group-Tags.

I would really appreciate if there was a way for it to work.

**Comment by alphapapa** [comment]
Yeah, I don't use group tags, and it seems that not many people do.  Since it's not a feature I use, I probably won't work on it anytime soon.  

I don't know how they're implemented internally.  Whether we can easily support them depends on whether the agenda sets group tags in text-properties--if not, we'll have to write code to get them from the source buffers, and whether that's difficult depends on how Org implements them.  If it's just a simple function call, it probably wouldn't be too difficult.

For anyone who wants to investigate this, here's what I recommend:

As a first step, pull up an agenda view that shows items that have group tags set, and use `C-u C-x =` to view the text properties of one of the items, to see if you can find the group tags in the properties.

If not, see if Org has a function to get specifically group tags from an item at point (in an Org buffer, not an Agenda buffer).

If not, then see if the standard Org tag-getting functions work with group tags.

And if the group tags aren't available in text properties, then we'll probably need to define a group-tags-specific selector in this package, to avoid making the regular one slower.

**Comment by k-jell** [comment]
It seems like tags are not part of the text properties. Instead they are are expanded as regular expressions when searching: [Matching tags](https://www.gnu.org/software/emacs/manual/html_node/org/Matching-tags-and-properties.html)

There is the function `org-tags-expand` which returns a regular expression matching all corresponding group tags. 

So when having defined a "maingroup" and "subgroup 1" + "subgroup 2", `(org-tags-expand "maingroup")` returns `"{\\<\\(?:maingroup\\|subgroup [12]\\)\\>}"`

I couldn't really figure out how `org-super-agenda--defgroup tag` works, but maybe this can help.

**Comment by alphapapa** [comment]
Thanks.  It looks like it may not be too difficult after all.  Please try evaluating this code and then running one of your views that should work with group tags:
```el
(org-super-agenda--defgroup tag
  "Group items that match any of the given tags.
Argument may be a string or list of strings."
  :let* ((target-tags (->> args (--map (org-group-tags it t)) -flatten)))
  :section-name (concat "Tags: " (s-join " OR " target-tags))
  :test (seq-intersection (org-super-agenda--get-tags item) target-tags 'cl-equalp))
```

**Comment by k-jell** [comment]
Unfortunately it doesn't work. `"Symbol’s function definition is void: org-group-tags"` is thrown when I try running an agenda view.

This seems to work for me:
```lisp
(org-super-agenda--defgroup tag
  "Group items that match any of the given tags.
Argument may be a string or list of strings."
  :let* ((target-tags (-flatten (cons args (org-tags-expand (car args) t)))))
  :section-name (concat "Tags: " (s-join " OR " args))
  :test (seq-intersection (org-super-agenda--get-tags item) target-tags 'cl-equalp))

```

**Comment by k-jell** [comment]
The previous code only worked while matching a single tag. This should also work with multiple tags:

```elisp
(org-super-agenda--defgroup tag
  "Group items that match any of the given tags.
Argument may be a string or list of strings."
  :let* ((target-tags (-flatten (cons args (mapcar (lambda (arg) (funcall #'org-tags-expand arg t)) args)))))
  :section-name (concat "Tags: " (s-join " OR " args))
  :test (seq-intersection (org-super-agenda--get-tags item) target-tags 'cl-equalp))
```

**Comment by ugurbolat** [comment]
> The previous code only worked while matching a single tag. This should also work with multiple tags:
> 
> ```emacs-lisp
> (org-super-agenda--defgroup tag
>   "Group items that match any of the given tags.
> Argument may be a string or list of strings."
>   :let* ((target-tags (-flatten (cons args (mapcar (lambda (arg) (funcall #'org-tags-expand arg t)) args)))))
>   :section-name (concat "Tags: " (s-join " OR " args))
>   :test (seq-intersection (org-super-agenda--get-tags item) target-tags 'cl-equalp))
> ```

This works :)
except for tag grouping with regular expressions such as Project : {P@.+}
do you have any ideas, how to solve that as well?

**Comment by alphapapa** [comment]
@ugurbolat This PR implementing group-tags support for org-ql shows how it can be done: https://github.com/alphapapa/org-ql/pull/146

**Comment by rickalex21** [comment]
Not sure what the status is on this or if @alphapapa  is going to add it but using @k-jell 
suggestion. You can get all the parent tags like this:

```emacs-lisp
;; You could use cl-flet but prob overkill (nconc final `()) is fine for now.
(org-super-agenda-groups
  (seq-map-indexed (lambda (x idx)
                  (let* ((count (+ idx 1))
                        (parent (car x))
                        (ph '(:underline t :foreground "green"))
                        (h2 '(:inherit  org-level-2))
                        (final `( :name ,parent :tag ,parent  )))
                        (pcase  parent
                                  ((pred (equal "Personal"))
                                    (nconc final `(:order 0 :face ,h2) ))
                                  ((pred (equal "Linux")) (nconc final `(:order 5)) )
                                  ((pred (equal "Emacs"))(nconc final `(:order 3 )) )
                                  ;; defaults
                                  (_ (nconc final `(:order ,idx))))
                                    final))
          (org-tag-alist-to-groups org-tag-alist)))
```
